system/handlers/formbuilder/Core.cfc

```diff
@@ -13,6 +13,11 @@ component {
 			event.notFound();
 		}
 
+		if ( !event.validateCsrfToken( rc.csrfToken ?: "" ) ) {
+			var csrfErrorMessage = translateResource( uri="formbuilder.item-types.csrf_token:invalid.csrf_token.error.message" );
+			setNextEvent( url=cgi.http_referer, persistStruct={ errorMessage=csrfErrorMessage } );
```

Comment:
We should probably be persisting the submission here also so that the user does not lose any data that they have filled in. i.e.

```cfc
var submission = event.getCollectionWithoutSystemVars(); // (raised from lines below)
// ...
var persist = submission;
persist.errorMessage = csrfErrorMessage;
// ...
```
```

Reply:
Thanks @DominicWatson  - the `getCollectionWithoutSystemVars()` method removes the `csrfToken` value before returning but I noticed it does a `duplicate()` beforehand so should be safe!!

Updated.